benchexec/tools/sv_benchmarks_util.py

```diff
@@ -40,3 +42,114 @@ def get_data_model_from_task(task, param_dict):
+    @return: List of witness files
+    """
+    witness_files, _ = _partition_input_files(task)
+    return witness_files
+
+
+def get_unique_witness(task):
+    """
+    This function returns the unique witness file from the task.
+    It raises an exception if there are multiple witness files.
+    The witness file is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: Unique witness file
+    """
+    witness_files = get_witness_input_files(task)
+    if len(witness_files) > 1:
+        raise UnsupportedFeatureException(
+            "Tool does not support tasks with more than one witness file"
+        )
+    return witness_files[0]
+
+
+def get_non_witness_input_files(task):
+    """
+    This function returns the non-witness input files from the task.
+    They consist of all files which do not match the witness file name.
+    The witness file is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: List of non-witness files
+    """
+    _, other_files = _partition_input_files(task)
+    return other_files
+
+
+def get_unique_non_witness_input_files(task):
+    """
+    This function returns the unique non-witness file from the task.
+    It raises an exception if there are multiple non-witness files.
+    Non-witness files consist of all files which do not match the witness file name.
+    The witness file is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: Unique non-witness file
+    """
+    other_files = get_non_witness_input_files(task)
+    if len(other_files) > 1:
+        raise UnsupportedFeatureException(
+            "Tool does not support tasks with more than one input file"
+        )
+    return other_files[0]
+
+
+def get_witness_options(options, task, witness_options):
+    """
+    This function returns the additional options to handle witnesses.
+    It checks if the witness is passed as an option or through the task definition.
+    If the witness is passed through both, it raises an exception.
+
+    @param options: List of existing options to see weather the witness
+        has already been passed to the tool
+    @param task: An instance of a task
+    @param witness_options: List of options which need to be set to handle witnesses
+        e.g. if the options are ["-w"], then the witness should be passed as "-w witness_file"
+        to the tool
+
+    @return: List of additional options to handle witnesses. They are constructed
+        based on witness_options. For example if witness_options is ["-w"], then
+        the return value will be ["-w", "witness_file"]
+    """
+
+    additional_options = []
+    if isinstance(task.options, dict) and "witness" in task.options.keys():
+        if not any(witness_option in options for witness_option in witness_options):
+            for witness_option in witness_options:
```

Comment:
Is it really intended that if `witness_options` is a list, the tool gets the witness passed with each of them? I don't expect that a tool needs the witness more than once on the command line. The documentation also does not explain this case.

I would suggest to replace with `witness_options` with a single value.

Reply:
For all tools except goblint this is correct. Goblint requires two parameters to be set to validate a correctness witness in version 2.0. Due to this I chose to implement the more general case which uses a list as `witness_options`.

Should I replace it with a single value and just call it twice inside goblint's toolinfo module?

Reply:
Maybe this is better yes.

> Goblint requires two parameters to be set to validate a correctness witness in version 2.0.

Because you are referring specifically to version 2.0 here: Does this mean that it will fail for non-2.0 witnesses? If differences between formats are important here, we need to think about how to handle this.

Reply:
> Maybe this is better yes.
Okay, will change the code to reflect this :ok_hand: 

> > Goblint requires two parameters to be set to validate a correctness witness in version 2.0.
> 
> Because you are referring specifically to version 2.0 here: Does this mean that it will fail for non-2.0 witnesses? If differences between formats are important here, we need to think about how to handle this.

If the options for goblint are different for non-2.0 witnesses it will fail. The difference between formats makes a difference, I tried to adjust the tool info modules based on the options given in the benchmark definition files. It would be easy to add support for other witnesses, by checking the witness version, for example based on the file ending or by parsing it and then adjust the options correspondingly.